enterprise/backend/src/metabase_enterprise/transforms/interface.clj

```diff
@@ -0,0 +1,31 @@
+(ns metabase-enterprise.transforms.interface)
+
+(defn- transform->transform-type
+  [transform]
+  (-> transform :source :type keyword))
+
+(defmulti source-db-id
+  "Return the source db id for a given transform."
+  {:added "0.47.0" :arglists '([transform])}
+  transform->transform-type)
+
+(defmulti target-db-id
+  "Return the target db id for a given transform."
+  {:added "0.47.0" :arglists '([transform])}
+  transform->transform-type)
+
+(defmulti execute!
+  "Execute a transform with an option."
+  {:added "0.47.0" :arglists '([transform options])}
+  (fn [transform _options]
+    (transform->transform-type transform)))
+
+(defmulti cancel!
+  "Execute a transform with an option."
+  {:added "0.47.0" :arglists '([transform options])}
+  (fn [transform _options]
```

Comment:
I'd probably have the interface target the 'run' rather than the transform at this level. The fact the http api takes a transform id I would hope changes as it introduces a kind of essential race condition. 

What about `(fn [transform run-id options])`?